### PR TITLE
tests: route: Convert legacy test to use ztest framework

### PR DESCRIPTION
**Pull request by punitvara**
This pacth makes use of ztest apis to make output unified with other
test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Comment by pfalcon** [comment]
Minor nitpick: commit message has a typo: "This pacth ...", should be "patch".

File: tests/net/route/src/main.c [hunk 1]
```diff
@@ -355,167 +336,129 @@ static bool populate_nbr_cache(void)
...
+	zassert_not_null(nbr, "Cannot add peer to neighbor cache");
 
 	k_sem_take(&wait_data, WAIT_TIME);
 
 	feed_data = false;
 
 	if (data_failure) {
 		data_failure = false;
-		return false;
+		zassert_true(0, NULL);
```
Comment (andrewboie) [inline tests/net/route/src/main.c]: please get rid of all these zassert_true(0, NULL) and replace with a direct test and a string explaining what the failure means

Comment (punitvara) [reply tests/net/route/src/main.c]: Sure. In some cases inside if loop, new value has been assigned to variable. In that case I can add failure message. Rest of cases I will replace with direct test and a string with failure message.

Comment (punitvara) [reply tests/net/route/src/main.c]: Fixed

**Pull request merged**